lightning/src/ln/channelmanager.rs

```diff
@@ -1203,6 +1204,26 @@ impl ChannelDetails {
 	}
 }
 
+/// Used to express the status of payments that are currently pending resolution.
+#[derive(Debug, PartialEq)]
+pub enum PendingPaymentDetails {
+	/// When a payment is still being sent and awaiting successful delivery.
+	Retryable {
+		/// Total amount (in msat) across all paths for this payment, not just the amount currently
+		/// inflight.
+		total_msat: u64
+	},
+	/// When a pending payment is fulfilled, we continue tracking it until all pending HTLCs have
+	/// been resolved. These payments will only cease being tracked upon receiving or handling
+	/// [`Event::PaymentSent`].
+	Fulfilled,
+	/// After a payment is explicitly abandoned by calling [`ChannelManager::abandon_payment`], they
+	/// are marked as abandoned until an [`Event::PaymentFailed`] event is generated. A payment
+	/// could also be marked as abandoned if pathfinding fails repeatedly and we will no longer
+	/// retry sending.
+	Abandoned,
```

Comment:
How useful are these without any other information? Should we at least include the payment hash in all variants?

Reply:
good point. I added it in 7efad83b